src/app/containers/PageHandlers/withHashChangeHandler/index.test.js

```diff
@@ -0,0 +1,41 @@
+import isHashChange from './isHashChange';
+
+describe('isHashChange', () => {
+  describe('prevent a rerender', () => {
+    it('should return true if the location.hash prop changes, preventing a re-render', () => {
```

Comment:
Wondering if there is a very niche scenario where someone clicks skip to content, and then does a client-side navigation.  In that scenario, `location.hash` changes, but also `location.pathname`.  Might this prevent a page from rerendering when it actually needs to?

We do barely any client side navigation at the minute, but it is being used between article pages - eg being on a `:service/articles/:id` page and clicking on another `:service/articles/:id` link will be done client side.

Reply:
yeah was able to reproduce this issue, eg:

1. update this file locally: https://github.com/bbc/simorgh/blob/latest/data/amharic/articles/czqverekrldo.json#L176 to be: `"locator": "/amharic/articles/czqverekrldo",` 
2. visit http://localhost:7080/afaanoromoo/articles/c4g19kgl85ko#content
3. click the link in the first paragraph

Reply:
Ahh, good catch. I can reproduce that as well. I'll have a further look to see if we can mitigate that for the pathname as well. Wasn't sure what the client-side routing situation was, but good to know its used in some areas.

Reply:
I've added in a `pathname` check and reworded the function slightly to better reflect its functionality.

Appears to resolve the issue you found, although slightly concerned that there are more of these gotchas if the URLs are used for much more than routing.

Reply:
LGTM 👍 only /articles links use CSR at the minute - rolling it out further has been on the backlog for a while but never made it to the top